src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -108,6 +108,9 @@ public function __invoke()
                         }
                         $token = file_get_contents($this->tokenFile);
                     }
+                    if (empty($token)) {
+                        throw new CredentialsException("InvalidIdentityToken from file: {$this->tokenFile}");
+                    }
```

Comment:
Hello @SamRemis,
thank you very much for your quick response.

I think this change just leads to another failure message?
In addition this should trigger a retry in my opinion.

I would prefer something like this at line 129 before calling assumeRoleWithWebIdentity():

```php
if (empty($assumeParams['WebIdentityToken'])) {
    throw new AwsException(
        "Empty InvalidIdentityToken",
        null,
        [
            'code' => 'InvalidIdentityToken',
        ]
    );
}
```

Best regards
Tobias

Reply:
Hi @tobiashaase,

I definitely agree that is the proper behavior; the reason I implemented it this way was just to get a quick fix in while I can do a bit more investigating.  I will need to look into how other SDKs are handling retries to make sure our behavior lines up and is appropriate.  I hesitate to copy the logic from below since the retries are meant for actual guzzle calls made by the client, so it's difficult to say how retries on reading the file should behave.  If there isn't a standard already established, I can make one :)

-Sam

Reply:
Ok - I understand. Thank very you.